src/js/components/DataTable/Body.js

```diff
@@ -85,67 +89,101 @@ const Body = forwardRef(
-                  size={size}
-                  active={active >= 0 ? active === index : undefined}
-                  onClick={
-                    onClickRow
-                      ? event => {
-                          // extract from React's synthetic event pool
-                          event.persist();
-                          const adjustedEvent = event;
-                          adjustedEvent.datum = datum;
-                          adjustedEvent.index = index;
-                          onClickRow(adjustedEvent);
-                        }
-                      : undefined
-                  }
-                  onMouseEnter={onClickRow ? () => setActive(index) : undefined}
-                  onMouseLeave={
-                    onClickRow ? () => setActive(undefined) : undefined
-                  }
-                  onFocus={onClickRow ? () => setActive(index) : undefined}
-                  onBlur={onClickRow ? () => setActive(undefined) : undefined}
-                >
-                  {(selected || onSelect) && (
-                    <TableCell background={background}>
-                      <CheckBox
-                        a11yTitle={`${
-                          isSelected ? 'unselect' : 'select'
-                        } ${primaryValue}`}
-                        checked={isSelected}
-                        disabled={!onSelect}
-                        onChange={() => {
-                          if (isSelected)
-                            onSelect(selected.filter(s => s !== primaryValue));
-                          else onSelect([...selected, primaryValue]);
+                <Fragment key={`${index.toString()}_expanded`}>
+                  <StyledDataTableRow
+                    key={primaryValue || index}
+                    ref={rowRef}
+                    size={size}
+                    active={active >= 0 ? active === index : undefined}
+                    onClick={
+                      onClickRow
+                        ? event => {
+                            // extract from React's synthetic event pool
+                            event.persist();
+                            const adjustedEvent = event;
+                            adjustedEvent.datum = datum;
+                            adjustedEvent.index = index;
+                            onClickRow(adjustedEvent);
+                          }
+                        : undefined
+                    }
+                    onMouseEnter={
+                      onClickRow ? () => setActive(index) : undefined
+                    }
+                    onMouseLeave={
+                      onClickRow ? () => setActive(undefined) : undefined
+                    }
+                    onFocus={onClickRow ? () => setActive(index) : undefined}
+                    onBlur={onClickRow ? () => setActive(undefined) : undefined}
+                  >
+                    {childComponent && (
+                      <ExpanderCell
```

Comment:
I'm wondering if the CheckBox cell for selecting the row should come first, since whether the row is expanded or not seems related to "within" whereas the selection state seems more about the whole thing.

Reply:
Addressed this.